test/LocatedTriplesTest.cpp

```diff
@@ -0,0 +1,704 @@
+
+  locatedTriplesPerBlock.erase(4, handles[1]);
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(3, 7, {{1, {1, 2}}, {2, {2, 0}}, {3, {0, 0}}, {4, {1, 1}}}));
+
+  locatedTriplesPerBlock.clear();
+
+  EXPECT_THAT(
+      locatedTriplesPerBlock,
+      m::allNums(0, 0, {{1, {0, 0}}, {2, {0, 0}}, {3, {0, 0}}, {4, {0, 0}}}));
+}
+
+TEST_F(LocatedTriplesTest, erase) {
+  using LT = LocatedTriple;
+  // Set up lists of located triples for three blocks.
+  auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock(
+      {LT{1, IT(10, 1, 0), false}, LT{1, IT(10, 2, 1), false},
+       LT{1, IT(11, 3, 0), true}, LT{2, IT(20, 4, 0), true},
+       LT{2, IT(21, 5, 0), true}, LT{4, IT(30, 6, 0), true},
+       LT{4, IT(32, 7, 0), false}});
+
+  auto handles = locatedTriplesPerBlock.add({LT{1, IT(15, 15, 15), false}});
+  EXPECT_THROW(locatedTriplesPerBlock.erase(5, handles[0]),
+               ad_utility::Exception);
+}
+
+// Test the method that merges the matching `LocatedTriple`s from a block into
+// an `IdTable`.
+TEST_F(LocatedTriplesTest, mergeTriples) {
+  using LT = LocatedTriple;
+
+  auto merge = [](size_t resultCols, size_t numIndexColumns, size_t resultRows,
+                  IdTable block, const LocatedTriplesPerBlock& locatedTriples) {
+    IdTable result(resultCols, ad_utility::testing::makeAllocator());
+    result.resize(resultRows);
+    locatedTriples.mergeTriples(1, std::move(block), result, 0,
+                                numIndexColumns);
+    return result;
+  };
+
+  // Merge the `LocatesTriples` into a block with 3 index columns.
+  {
+    IdTable block = makeIdTableFromVector({
+        {1, 10, 10},  // Row 0
+        {2, 15, 20},  // Row 1
+        {2, 15, 30},  // Row 2
+        {2, 20, 10},  // Row 3
+        {2, 30, 20},  // Row 4
+        {3, 30, 30}   // Row 5
+    });
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock({
+        LT{1, IT(1, 5, 10), true},    // Insert before row 0
+        LT{1, IT(1, 10, 10), false},  // Delete row 0
+        LT{1, IT(1, 10, 11), true},   // Insert before row 1
+        LT{1, IT(2, 11, 10), true},   // Insert before row 1
+        LT{1, IT(2, 30, 10), true},   // Insert before row 4
+        LT{1, IT(2, 30, 20), false},  // Delete row 4
+        LT{1, IT(3, 30, 25), false},  // Delete non-existent row
+        LT{1, IT(3, 30, 30), false},  // Delete row 5
+        LT{1, IT(4, 10, 10), true},   // Insert after row 5
+    });
+    IdTable resultExpected = makeIdTableFromVector({
+        {1, 5, 10},   // LT 1
+        {1, 10, 11},  // LT 2
+        {2, 11, 10},  // LT 3
+        {2, 15, 20},  // orig. Row 1
+        {2, 15, 30},  // orig. Row 2
+        {2, 20, 10},  // orig. Row 3
+        {2, 30, 10},  // LT 4
+        {4, 10, 10}   // LT 7
+    });
+
+    auto merged =
+        merge(resultExpected.numColumns(), 3, resultExpected.numRows(),
+              std::move(block), locatedTriplesPerBlock);
+    EXPECT_EQ(merged, resultExpected);
```

Comment:
If you use `EXPECT_THAT(merged, ElementsAreArray(expected))` you get better diagnostics.